### PR TITLE
New GetAcctEntitlements() api

### PR DESCRIPTION
**Pull request by rchowinfoblox**
Other changes:
- Refactor/CustomQueryStream to directly decode OPA response stream without reading entire response into tmp buffer
- Fixed OPA input for explicit decision-document
- Added middleware cmdline utility for testing purposes
- General code cleanup

### CODE REVIEW

File: grpc_opa/acct_entitlements.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+type AcctEntitlementsType map[string]map[string][]string
+
+// AcctEntitlementsApiResult is the data type returned by OPA RESTAPI query to acct_entitlements_api
+type AcctEntitlementsApiResult struct {
+	Loggr  *logrus.Logger        `json:"-"`
+	Result *AcctEntitlementsType `json:"result"`
+}
+
+// UnmarshalJSON implements json.Unmarshaler interface for AcctEntitlementsApiResult
+func (u *AcctEntitlementsApiResult) UnmarshalJSON(rawBytes []byte) error {
```
Comment (AnPiakhota-Infoblox) [inline grpc_opa/acct_entitlements.go:31]: This is too fragile. We can get more readable code using the following struct: 

```go
var results = []string{
`{ "result": {
	"acct1": { "svc1a": [ "feat1a1", "feat1a2" ] },
	"acct2": { "svc2a": [ "feat2a1", "feat2a2" ],
		       "svc2b": [ "feat2b1", "feat2b2" ] }
}}`,
// ---
`{ "result": null }`,
// ---
`{ "result": {
	"acct1": { "svc1a": [ "feat1a1", "feat1a2" ] },
	"acct2": null
}}`,
}

type result struct {
	Result map[string]map[string]map[string]interface{}
}

func main() {
	for i, res := range results {
		var r result
		err := json.Unmarshal([]byte(res), &r.Result)
		printj(i, r, err)
	}
}
```

Please play it [here](https://go.dev/play/p/fw1SUwS6mkU).

File: grpc_opa/acct_entitlements.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+	DefaultAcctEntitlementsApiPath = "v1/data/authz/rbac/acct_entitlements_api"
+)
+
+// AcctEntitlementsType is a convenience data type, returned by GetAcctEntitlements()
+// (map of acct_id to map of service to array of features)
+type AcctEntitlementsType map[string]map[string][]string
+
+// AcctEntitlementsApiResult is the data type returned by OPA RESTAPI query to acct_entitlements_api
+type AcctEntitlementsApiResult struct {
+	Loggr  *logrus.Logger        `json:"-"`
```
[the remark this answers is not in the document]
Comment (rchowinfoblox) [reply grpc_opa/acct_entitlements.go]: Removed, found out custom unmarshaler is actually not required

File: grpc_opa/acct_entitlements.go (hunk 1 above)
Comment (rchowinfoblox) [reply grpc_opa/acct_entitlements.go]: yes I found out custom unmarshaler is actually not required, i've removed it.

**Review by arjunbhandage — COMMENTED** [review]
looks good. A single api for single and multi/all accounts. I will let @daniel-garcia approve this

**Review by AnPiakhota-Infoblox — APPROVED** [review]

**Pull request merged**